compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/UIKitTextInputService.uikit.kt

```diff
@@ -298,6 +301,32 @@ internal class UIKitTextInputService(
             sendEditCommand(FinishComposingTextCommand())
         }
 
+        override fun positionFromPosition(position: Long, offset: Long): Long {
+            val text = getState()?.text ?: return 0
+
+            if (position + offset >= text.lastIndex + 1) {
+                return (text.lastIndex + 1).toLong()
+            }
+            if (position + offset <= 0) {
+                return 0
+            }
+            var currentOffset = offset
+            var resultPosition = position.toInt()
+
+            val it = BreakIterator.makeCharacterInstance()
+            it.setText(text)
+
+            while (currentOffset > 0) { // Move forward
+                resultPosition = it.following(resultPosition)
+                currentOffset--
+            }
+            while (currentOffset < 0) { // Move backward
```

Comment:
Two such loops in a row doesn't look correct. Should it choose only one direction based on offset?

Why do you need it at all? `preceding`/`following` should do it inside

Reply:
We need to make N steps in some direction.
`N = offset` in this case.
And we need to call both `following` and `preceding` in different directions.

This while loop is very simple and easy to read. I want to leave it as is.